src/library/scala/collection/package.scala

```diff
@@ -77,4 +77,12 @@ package object collection {
       if(t.isEmpty) None
       else Some(t.init -> t.last)
   }
+
+  /** Common message for IndexOutOfBounds exceptions with a known max index */
+  private[collection] def indexOutOfBoundsMessage(index: Int, max: Int): String = 
+    s"$index is out of bounds (min 0, max ${max})"
+
+  /** Common message for IndexOutOfBounds exceptions with an unknown max index. */
+  private[collection] def indexOutOfBoundsMessage(index: Int): String = 
+    s"$index is out of bounds (min 0, max unknown)"
```

Comment:
Since every call site follows up with a `new IndexOutBoundsException`, shouldn't we also include the latter inside the method? That would definitely reduce the bytecode more.

Perhaps we should even include the `throw` in it as well and return `Nothing`? (That may have an adverse effect on the JVM, as we wouldn't statically know that the rest is dead code from there.)

Reply:
That's a good point. Although given that it was deeped acceptable for `Statics.ioobe`, I assume it should be fine here as well.

Reply:
I think so too.